### PR TITLE
【PaddlePaddle Hackathon 3 No.15】为 Paddle 新增 count_nonzero

### PR DESCRIPTION
**Pull request by thunder95**
### PR types

New features 
### PR changes

 APIs
### Describe

完成第三期第15项目开发任务: https://github.com/PaddlePaddle/Paddle/issues/44073#task15
计算输入中非零元素的个数。此任务的目标是在 Paddle 框架中，新增 count_nonzero API，调用路径为：paddle.count_nonzero 和 Tensor.count_nonzero。

RFC设计文档: https://github.com/PaddlePaddle/community/pull/167 
中文文档：https://github.com/PaddlePaddle/docs/pull/5085

### CODE REVIEW

**Review by luotao1 — DISMISSED** [review]
LGTM，看下CI的情况

**Comment by Ligoml** [comment]
需要补充中文API文档到 [docs](https://github.com/PaddlePaddle/docs) repo 哈

**Comment by thunder95** [comment]
> 需要补充中文API文档到 [docs](https://github.com/PaddlePaddle/docs) repo 哈

已添加　@Ligoml

**Comment by luotao1** [comment]
> Sorry to inform you that https://github.com/PaddlePaddle/Paddle/commit/3f8427ccf8276ed5259c86737bf8b8d541b9f31c's CIs have passed for more than 7 days. To prevent PR conflicts, you need to re-run all CIs manually.

@thunder95 由于超过7天了，请手动rerun下失败的CI吧。

**Comment by Ligoml** [comment]
> > 需要补充中文API文档到 [docs](https://github.com/PaddlePaddle/docs) repo 哈
> 
> 已添加　@Ligoml

中文文档有一点小问题，英文文档目前预览无法正常生成，需要重新merge一下develop分支，待 PR-CI-Build 通过后再查看

**Comment by Ligoml** [comment]
image
这个模板注意不要删，容易被CheckPRTemplate卡住

**Review by Ligoml — DISMISSED** [review]
LGTM for docs

**Comment by thunder95** [comment]
@Ligoml 奇怪 这个模板我应该没删过的

File: python/paddle/fluid/tests/unittests/test_count_nonzero_api.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+            if isinstance(axis, list):
+                axis = list(axis)
+                if len(axis) == 0:
+                    axis = None
+            x_tensor = paddle.to_tensor(x, stop_gradient=False)
+            y = paddle.count_nonzero(x_tensor, axis, keepdim)
+            dx = paddle.grad(y, x_tensor)[0].numpy()
+            self.assertEqual(
+                np.allclose(np.sum(np.ones_like(x, dtype=np.int64)), dx.sum()),
+                True)
```
Comment (jeff41404) [inline python/paddle/fluid/tests/unittests/test_count_nonzero_api.py]: Is the gradient of x obtained by this code correct in math? or x should have gradient?

Comment (thunder95) [reply python/paddle/fluid/tests/unittests/test_count_nonzero_api.py]: Thanks for reminding me, Mr. Gao. Here this count operator was mistaken for sum operator by me. According to gradient law, x is not supposed to have gradient. @jeff41404

**Review by jeff41404 — APPROVED** [review]
LGTM

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Review by Ligoml — APPROVED** [review]
LGTM for docs

**Pull request merged**